compiler/env/OMRAOTMethodHeader.cpp

```diff
@@ -0,0 +1,116 @@
+/*******************************************************************************
+ * Copyright (c) 2020, 2020 IBM Corp. and others
+ *
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at http://eclipse.org/legal/epl-2.0
+ * or the Apache License, Version 2.0 which accompanies this distribution
+ * and is available at https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the
+ * Eclipse Public License, v. 2.0 are satisfied: GNU General Public License,
+ * version 2 with the GNU Classpath Exception [1] and GNU General Public
+ * License, version 2 with the OpenJDK Assembly Exception [2].
+ *
+ * [1] https://www.gnu.org/software/classpath/license.html
+ * [2] http://openjdk.java.net/legal/assembly-exception.html
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-exception-2.0 OR LicenseRef-GPL-2.0 WITH Assembly-exception
+ *******************************************************************************/
+#include "env/AOTMethodHeader.hpp"
+#include <stdlib.h>
+#include <string.h>
+
+TR::AOTMethodHeader* OMR::AOTMethodHeader::self()
+   {
+   return reinterpret_cast<TR::AOTMethodHeader*> (this);
+   }
+
+void* OMR::AOTMethodHeader::serialize()
+   {
+   uintptr_t allocSize = self()->sizeOfSerializedVersion();
+
+   // This buffer contains the message to the cache.
+   // This memory needs to be released after the
+   // header is used in the cache. 
+   uint8_t* buffer = (uint8_t*) malloc(allocSize);
```

Comment:
Hi, @mstoodle  thank you for the review. I have came up with a simplified solution, it will be shipped with the next commit. Can you please comment how does that look as a fix?

Reply:
I will push the changes soon.

Reply:
Hello @mstoodle, how do the changes look to you: A storage class implementing AOTStorageInterface is responsible for allocating memory, and it's the storage classes responsibility to free the memory once the AOT item was persisted. The simple changes to the interface is in the PR already.